lib/vmpooler/providers/gce.rb

```diff
@@ -194,6 +194,7 @@ def create_vm(pool_name, new_vmname)
           # Assume all pool config is valid i.e. not missing
           client = ::Google::Apis::ComputeV1::Instance.new(
             name: new_vmname,
+            hostname: "#{new_vmname}.#{domain || global_config[:config]['domain']}",
```

Comment:
There is a 3rd case where both domain and global domain is not set and we should then just use the hostname.
Something like:
```suggestion
          append_domain = domain || global_config[:config]['domain']
          if append_domain
            fqdn = "#{new_vmname}.#{append_domain}"
          else
            fqdn = new_vmname
          end
          # Assume all pool config is valid i.e. not missing
          client = ::Google::Apis::ComputeV1::Instance.new(
            name: new_vmname,
            hostname: fqdn,
```

Reply:
Ah good call, but If the domain or global domain is not set, wouldn't we want to either omit the hostname key or set it to nil, so that it defaults to the expected VM_NAME.c.PROJECT_ID.internal name (which was the previous behavior)?

Reply:
perhaps yes that would make more sense. I do not know if setting `hostname` to `nil` in Intance.new() will do that, we need to check.